### DISCUSSION TITLE
Using ThreeCPP from outside

### DISCUSSION THREAD

**Discussion by seanbutler** [markaren/threepp#152] [General]
Hello, 

This is a super great project as three.js is so popular and widely known. 

I'd like to use it as a high level library over opengl in my own projects.
I can see the test folder with the example of building a simple executable from within the threepp project.
However am struggling to understand how to build a separate CMake project which depends on or accesses threepp but us unrelated. Do you  have any advice (on ubuntu)?

Thanks, 
Sean

**Comment by markaren** [comment]
You can use vcpkg with a custom registry. See: https://github.com/markaren/threepp/tree/master/tests/threepp_vcpkg_test

That folder is a standalone project with access to some version of threepp (retreived using vcpkg).

The custom registry is located at https://github.com/Ecos-platform/vcpkg-registry and is maintaned by myself.
You can see the port file here: https://github.com/Ecos-platform/vcpkg-registry/blob/main/ports/threepp/portfile.cmake

Otherwise, using threepp as a CMake subproject (e.g. git submodule) should be possible.

**Comment by markaren** [comment]
I added a README to the demo project, which should make things clearer (dev branch).

**Reply by seanbutler** [reply]
Thank You. I'll take a look!

**Comment by josefgraus** [comment]
I also just submodule threepp in git then in my CMakeLists.txt in the parent directory of where the submodule is located:
```
set(THREEPP_BUILD_EXAMPLES OFF)
set(THREEPP_BUILD_EXAMPLE_PROJECTS OFF)
set(THREEPP_BUILD_TESTS OFF)
set(BUILD_SHARED_LIBS OFF)
add_subdirectory(threepp)
```
That will add a `threepp` cmake target to your project that you can link against in your project configuration, something like:
```
target_link_libraries(${PROJECT_NAME} PRIVATE threepp)
```

**Reply by josefgraus** [reply]
Suppose I should mention you need to resolve the other dependencies yourself via this method. I used Conan 2.0. The recommended is vcpkg. You can also submodule all the dependencies from their various sources and add them to your cmake project in your own fashion. That last option is probably the most painful though.
